apps/site/assets/ts/stop/components/StopPageRedesign.tsx

```diff
@@ -1,10 +1,81 @@
-import React, { ReactElement } from "react";
+import React, { ReactElement, useEffect, useState } from "react";
+import Pill from "./Pill";
+import { Mode } from "../../__v3api";
+import { uniqueId } from "lodash";
+
+export const BUS = "bus";
+export const SUBWAY = "subway";
+export const FERRY = "ferry";
+export const COMMUTER_RAIL = "commuter_rail";
```

Comment:
I was thinking of putting these in their own Mode model file.  Right now Mode typing lives in the `v3api` data file.  Those file types don't support variable declarations/exporting, so it would need to move as well.

Reply:
Would something like this in the v3api file be what you're getting at?

```typescript
export enum ModeEnum {
  commuter_rail,
  subway,
  bus,
  ferry
}
 
export type Mode = keyof typeof ModeEnum;
// export type Mode = "commuter_rail" | "subway" | "bus" | "ferry";
```

Then you can use the enum value as your constant (e.g. your `SUBWAY` would be `ModeEnum.subway` here.

Reply:
So the problem I was running into was `v3api` is a [declaration file](https://www.typescriptlang.org/docs/handbook/2/type-declarations.html#dts-files) which has very specific rules about what can be in them.  The only thing allowed is types.
The enum would have to be in its own typescript file (which would have to be imported inline via `import('./ModeEnum').BUS`)
It seems hacky but it can be done.

Reply:
aahhh I see how that the only place we use enum are indeed in .ts files. So much for that. Maybe there's a less wacky way 🤔